docs/csharp/whats-new/csharp-8.md

```diff
@@ -0,0 +1,314 @@
+---
+title: What's New in C# 8.0 - C# Guide
+description: Get an overview of the new features available in C# 8.0. This article is up-to-date with preview 2.
+ms.date: 02/12/2019
+---
+# What's new in C# 8.0
+
+> [!NOTE]
+> This article was last updated for C# 8.0 preview 2.
+
+There are many enhancements to the C# language that you can try out already with preview 2. The new features added in preview 2 are:
+
+- [Pattern matching enhancements](#more-patterns-in-more-places):
+  * [Switch expressions](#switch-expressions)
+  * [Property patterns](#property-patterns)
+  * [Tuple patterns](#tuple-patterns)
+  * [Positional patterns](#positional-patterns)
+- [Using declarations](#using-declarations)
+- [Static local functions](#static-local-functions)
+- [Disposable ref structs](#disposable-ref-structs)
+
+The following language features first appeared in C# 8.0 preview 1:
+
+- [Nullable reference types](#nullable-reference-types)
+- [Asynchronous streams](#asynchronous-streams)
+- [Ranges and indices](#ranges-and-indices)
+
+The remainder of this article briefly describes these features. Where in-depth articles are available, links to those tutorials and overviews are provided.
+
+## More patterns in more places
+
+**Pattern matching** gives tools to provide shape-dependent functionality across related but different kinds of data. C# 7.0 introduced syntax for type patterns and constant patterns by using the `is` expression and the `switch` statement. These features represented the first tentative steps toward supporting programming paradigms where data and functionality live apart. As the industry moves toward more microservices and other cloud-based architectures, other language tools are needed.
```

Comment:
I don't think perf work has been completed. So I won't address this yet. I think it will be similar perf, but I'll verify as we get closer to release.